### ISSUE TITLE
support RGBA color values

### ISSUE THREAD

**Issue by lidaof** [3dmol/3Dmol.js#513]
Is it possible to support RGBA color values? I am trying to color one chain with a color function using resi to assign colors?
I would like resi not selected to be a little bit transparent, tried using RGBA or hex code with alpha not working.

Thanks.

**Comment by dkoes** [comment]
opacity can be set as a separate parameter, like here:
https://3dmol.csb.pitt.edu/tests/auto/generate_test.cgi?test=test98

But I agree that also supporting RGBA values would be convenient.

**Comment by dkoes** [comment]
What would be your preferred syntax for specifying rgba in one value? It doesn't seem to be common to use hex.

**Comment by lidaof** [comment]
Agree, i think css `rgba` way is more common. Thanks for consideration.

**Comment by lidaof** [comment]
thanks you @dkoes

**Comment by lidaof** [comment]
Hi @dkoes, rethinking about this issue, is it possible to allow `colorfunc` to return an object with opacity instead of just a color string? Thank you in advance for consideration.

**Comment by jaxmatrix** [comment]
I have another thought on this lets not change the way color works but rather the way the shader is processing the color. Inside the fragment shaders we can change `varying vec3 vColor` to `varying vec4 vColor` and `gl_FragColor = vec4( vColor.x, vColor.y, vColor.z, vColor.w*opacity)` this will accommodate both `rgba` colors along with `opacity`. And in the renderer loading of the color uniform can be done using `this._gl.uniform3f(uniformLoc, uniformVal.r, uniformVal.g, uniformVal.b, uniformVal.a ? uniformVal.a : 1)`. These are just my initial thoughts, I still don't know what other changes might be required.
